File: model_training/3_1CopernicusMarine.py
import copernicusmarine as cm

cm.login
username = ""
password = ""

#For temperature want to use GLOBAL_MULTIYEAR_PHY_001_030
#For Net primary production want to use: GLOBAL_MULTIYEAR_BGC_001_033

cm.subset(
  dataset_id="cmems_mod_glo_phy_my_0.083deg_P1M-m",
  dataset_version="202311",
  variables=["so", "thetao"],
  minimum_longitude=-12.01498103062979,
  maximum_longitude=13.645282778488214,
  minimum_latitude=47.183697679290205,
  maximum_latitude=61.77604047194526,
  start_datetime="1999-01-01T00:00:00",
  end_datetime="2021-06-01T00:00:00",
  minimum_depth=0.49402499198913574,
  maximum_depth=0.49402499198913574,
)

# cm.subset(
#   dataset_id="cmems_mod_glo_bgc_my_0.083deg-lmtl_PT1D-i",
#   dataset_version="202211",
#   variables=["npp"],
#   minimum_longitude=-12.01498103062979,
#   maximum_longitude=13.645282778488214,
#   minimum_latitude=47.183697679290205,
#   maximum_latitude=61.77604047194526,
#   start_datetime="1999-01-01T00:00:00",
#   end_datetime="2022-12-31T00:00:00",
# )

File: model_training/3_1CopernicusMarine.R
library(reticulate) #reticulate package allows for python usage in R 
virtualenv_create("mbo-proj",force=FALSE) #create a virtual environment to install packages in
py_install("copernicusmarine",envname = "mbo-proj") #the copernicusmarine package allows CMEMS downloads
virtualenv_list() #Check the list of available environments
use_virtualenv("mbo-proj") #Load the environment

# More information on the reticulate package on: https://rstudio.github.io/cheatsheets/reticulate.pdf
# 
# Overview of the functions can be found at: https://help.marine.copernicus.eu/en/collections/9080063-copernicus-marine-toolbox
# 
# How to configure the credentials can be found at: https://help.marine.copernicus.eu/en/articles/8185007-copernicus-marine-toolbox-credentials-configuration
# Works in the terminal. Needs to be done only once. 

bbox <-sf::st_bbox(spatial_extent)
xmin <- bbox[[1]]
xmax <- bbox[[3]]
ymin <- bbox[[2]]
ymax<- bbox[[4]]


cm <- import("copernicusmarine")

cm$subset(
  dataset_id="cmems_mod_glo_phy_my_0.083deg_P1M-m",
  variables=c("so", "thetao"),
  minimum_longitude=xmin,
  maximum_longitude=xmax,
  minimum_latitude=ymin,
  maximum_latitude=ymax,
  start_datetime=date_start,
  end_datetime=date_end,
  minimum_depth=0.49402499198913574,
  maximum_depth=0.49402499198913574,
  output_directory= envdir,
  output_filename="tempsal.nc",
  overwrite_output_data = TRUE,
  force_download=TRUE)

cm$subset(
  dataset_id="cmems_mod_glo_bgc_my_0.083deg-lmtl_PT1D-i",
  variables=list("npp"),
  minimum_longitude=xmin,
  maximum_longitude=xmax,
  minimum_latitude=ymin,
  maximum_latitude=ymax,
  start_datetime=date_start,
  end_datetime=date_end,
  minimum_depth=0.5057600140571594,
  maximum_depth=0.5057600140571594,
  output_directory= envdir,
  output_filename="npp.nc",
  overwrite_output_data = TRUE,
  force_download=TRUE)

